src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -635,3 +635,21 @@ def _configure_security_policy_storage_params(cmd):
                options_list=('--end-ip-address',),
                help='The end IP address of the firewall rule. Must be IPv4 format. Use value'
                ' \'0.0.0.0\' to represent all Azure-internal IP addresses.')
+
+
+#####
+#           sql server vnet-rule
+#####
+
+
+with ParametersContext(command='sql server vnet-rule') as c:
+    # Help text needs to be specified because 'sql server firewall-rule update' is a custom
+    # command.
+    c.argument('server_name',
+               options_list=('--server', '-s'))
+
+    c.argument('virtual_network_rule_name',
+               options_list=('--name', '-n'))
+
+    c.argument('virtual_network_subnet_id',
+               options_list=('--vnet-subnet-id', '-v'))
```

Comment:
The signature should look very similar to `az storage account network-rule add`.  Namely, there should be parameters `--subnet` which takes the name or ID of a subnet and `--vnet-name` which takes the name of a VNET (used in conjunction with `--subnet`).

Reply:
We have a lot of difference with the way vnet is implemented between storage and Sql. Because of this, we have not maintained storage consistency in any of the other user facing api's. I would rather have consistency with our other api's (powershell, .net, ARM and Portal) than try to get storage consistency. Making this change will break that pattern.

Reply:
This is about consistency with the entire CLI--the storage commands were just an easy example since they are related to the same feature. NICs, load balancers... anywhere in the CLI that a subnet ID is required by the service or SDK, the CLI exposes this as `--subnet NAME_OR_ID --vnet-name NAME`.

Reply:
I could rename the vnet-subnet-id to subnet-id. But I don't understand the point of asking the user for a variable called vnet-name since we never use it in our code.

Reply:
Here's the code you can reuse from the storage commands than handles the whole name/ID thing:
https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py#L790